security_privacy_ln.asciidoc

```diff
@@ -0,0 +1,614 @@
+Generally, this is achieved by the attacker bombarding a resource with requests, which are indistinguishable from legitimate queries.
+The attacks seldom result in the target suffering financial loss aside from the opportunity cost of their service being down and are merely intended to grief the target.
+
+Typical mitigations for denial-of-service attacks require authentication for requests to separate legitimate users from malicious ones or to incur a trivial cost to regular users but will act as a sufficient deterrent to an attacker launching requests at scale.
+Anti-denial-of-service measures can be seen everywhere on the internet - websites apply rate limits to ensure that no one user can consume all of their server's attention, film review sites require login authentication to keep angry r/prequelmemes members at bay, and API keys are purchased to allow a limited number of hits.
+
+=== DoS in Bitcoin
+
+In Bitcoin, the bandwidth that nodes use to relay transactions and the space that they avail to the network in the form of their mempool are publicly available resources.
+Any node on the network can consume bandwidth and mempool space by sending a valid transaction.
+If this transaction is mined in a valid block, they will pay transaction fees, which adds a cost to using these shared network resources.
+
+In the past, the Bitcoin network faced an attempted denial-of-service attack where attackers spammed the network with low-fee transactions.
+Many of these transactions were not selected by miners due to their low transaction fees, so the attackers could consume network resources without paying the fees.
+To address this issue, a minimum transaction relay fee that set a threshold fee that nodes require to propagate transactions was set.
+This measure largely ensured that the transactions that consume network resources will eventually pay their chain fees.
+The minimum relay fee is acceptable to regular users but would hurt attackers financially if they tried to spam the network.
+While some transactions may not make it into valid blocks during high fee environments, these measures have largely been effective at disincentivizing this type of spam.
+
+=== DoS in Lightning
+
+Similarly to Bitcoin, the Lightning Network charges fees for the use of its public resources, but in this case, the resources are public channels, and the fees come in the form of routing fees.
+The ability to route payments through nodes in exchange for fees provides the network with a large scalability benefit - nodes that are not directly connected can still transact - but it comes at the cost of exposing a public resource that must be protected against DoS attacks.
+
+When a Lightning node forwards a payment on your behalf, it uses bandwidth to update its commitment transaction, and the amount of the payment is reserved in their channel balance until it is settled or failed.
```

Comment:
Unclear if "bandwidth" here refers to payment bandwidth, or actual data bandwidth, works in either interpretation though IMO.